notebooks/Notebook1.md

```diff
@@ -1,34 +1,61 @@
+chart image:
+
+<img width="602" alt="Screen Shot 2023-03-30 at 7 55 34 PM" src="https://user-images.githubusercontent.com/97985236/229011324-73a37a60-9a6c-4df1-becb-0fb27c6c819e.png">
+                                                                 
+<img width="609" alt="Screen Shot 2023-03-30 at 7 56 25 PM" src="https://user-images.githubusercontent.com/97985236/229011458-dc9f8c81-b94c-42b0-9ea6-41e305c15878.png">
+
+image of possibe inputs:
+
+<img width="233" alt="Screen Shot 2023-03-30 at 7 57 16 PM" src="https://user-images.githubusercontent.com/97985236/229011588-d87b5cbd-4055-46f5-9c0d-92ce86e1a7dc.png">
+
+Continuing this week I got ahead by planning out and implementing the data structure that would represent a spreadsheet of a football game. I began by drawing up a layout of a hashmap with a possible representation of a player that may be stored within it (see images below). Then I wrote down some notes/drawings of things that I want to make sure get implemented with this data structure such as having the ability to update a player or add a player to the hashmap. (see images below). Next I began my implementation by building a game class. This class houses the hashmap that has a string key representing the type of play and an arraylist of players to be stored for each hash. This was straight forward and I had no problems([link to code here](https://github.com/hmc-cs111-spring2023/BBeckerArtifact/blob/main/src/Game/Game.java)). Next I began building an abstract parent class called player. The goal of this class is to hold all possible variables, methods, getters, constructors that may be used in a given player type (subclasses)([link to code here](https://github.com/hmc-cs111-spring2023/BBeckerArtifact/blob/main/src/Player/Player.java)). This was straightforward but took some experimenting with the subclasses. Next I build a subclass for each type of player (9 total). Each subclass represents a different type of player and each type's constructor is slightly different ([link to code here](https://github.com/hmc-cs111-spring2023/BBeckerArtifact/tree/main/src/Player)). But the creation of these classes went very smoothly. Next I worked on a method in the game class that would allow me to add a player to a specified hash given the string of the hash and the player to be added. The add method is very similar to the default add method but just tailored towards data structure and the player class. Finally to end the week I build a function called setPlayer that allows for specific attributes of a player to be updated. Say you wanted to update the number of yards a rushing player rushed for you could do so with this method. The challenge of this method was that each player type needed a slightly different implementation. Overall this was not hard but tedious. ([link for code](https://github.com/hmc-cs111-spring2023/BBeckerArtifact/tree/main/src)) 
```

Comment:
Great progress with the implementation of these data structures and classes. I liked that you noted that the work was tedious despite not being complex, is there any way that you think you could have implemented these that would have been less tedious? I do not have any suggestions based on looking at it but I think it could be good to reflect on the code you wrote, especially when you found something repetitive to see if improvements can be made. I wouldn't commit too much time to this since it likely won't impact the functionality and hence the resulting DSL but if there was something that you found tedious when implementing this, you could try to learn from that experience and thus the next time you are creating something with a similar class structure you could do so more quickly. Thanks for linking the code you were referring to, it made it easy to look back at each step you discussed!